kevm-pyk/src/kevm_pyk/kevm.py

```diff
@@ -360,3 +362,58 @@ def accounts(accts: list[KInner]) -> KInner:
             else:
                 wrapped_accounts.append(acct)
         return build_assoc(KApply('.AccountCellMap'), KLabel('_AccountCellMap_'), wrapped_accounts)
+
+    def pretty_print(self, kast: KAst, *, unalias: bool = True) -> str:
+        if unalias and isinstance(kast, KInner):
+            kast = undo_aliases(self.definition, kast)
+        if isinstance(kast, KInner):
+            kast = KEVM._int_token_to_hex(kast)
+        return super().pretty_print(kast, unalias=False)
+
+    @staticmethod
+    def _int_token_to_hex(kast: KInner) -> KInner:
+        def _helper(_kast: KInner) -> KInner:
+            if type(_kast) is KToken and _kast.sort == INT:
+                return KToken(hex(int(_kast.token)), INT)
+            if type(_kast) is KToken and _kast.sort == BYTES:
+                return KToken('0x' + pretty_bytes(_kast).hex(), BYTES)
+            return _kast
```

Comment:
Maybe we only want to do this for numbers taht are bigger than 256 when it's an `INT`? If the user writes `3` in their code, and it ends up on the stack, then they want to see `3` directly, not the hex encoding of it.